### ISSUE TITLE
failed to load vite wasm

### ISSUE THREAD

**Issue by zavx0z** [oven-sh/bun#7146]
### What version of Bun is running?

1.0.11+f7f6233ea

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

install vite-plugin-wasm and configure

```
import { sveltekit } from '@sveltejs/kit/vite';
import { defineConfig } from 'vite';
import wasm from "vite-plugin-wasm";

export default defineConfig({
	plugins: [wasm(), sveltekit()]
});
```

### What is the expected behavior?

![image](https://github.com/oven-sh/bun/assets/7262706/1abeee80-a0f8-4709-9843-0e2b59416529)

**Comment by Menci** [comment]
Hi! I'm the developer of the package `vite-plugin-wasm`. Here is a MWE for this issue:

<img width="796" alt="image" src="https://github.com/oven-sh/bun/assets/11341955/936961c2-31c1-428d-89d6-a9b578a0276a">

**tl; dr** Node.js is ignoring `__esModule` and treating the whole `exports` object as an CJS file's default export. So I need to write `import(cjsFile).default.default` to get the `"default"` key of its `exports` object while Bun's behavior is like TypeScript's `esModuleInterOp`.

It's my first time to Bun and I'm not sure whether it's by design or a bug to have such inconsistent behavior to Node.js. But anyway I'll workaround it in my library.
